source/tools/line_count/src/main.rs

```diff
@@ -1305,19 +1305,11 @@ impl<'f> Visitor<'f> {
     }
 
     fn fn_code_kind(&self, kind: CodeKind) -> CodeKind {
-        if self.in_state_machine_macro > 0 {
-            kind.join_prefer_left(CodeKind::Spec)
-        } else {
-            kind
-        }
+        if self.in_state_machine_macro > 0 { kind.join_prefer_left(CodeKind::Spec) } else { kind }
     }
 
     fn mode_or_trusted(&self, kind: CodeKind) -> CodeKind {
-        if self.trusted > 0 {
-            CodeKind::Trusted
-        } else {
-            kind
-        }
+        if self.trusted > 0 { CodeKind::Trusted } else { kind }
```

Comment:
On suggestion by @jaylorch, pinging @jaybosamiya-ms about this code where `vargo fmt` and `verusfmt` disagree about the right way to format.  This seems related to the fact that `vargo fmt` passes `--config version=Two` to `rustfmt`.

Reply:
Huh, looks like `vargo fmt` for some reason is passing extra config to `rustfmt` as arguments, rather than use the existing `rustfmt.toml` file (https://github.com/verus-lang/verus/blob/main/source/rustfmt.toml); indeed it even includes the config but comments it out.

Verusfmt is already aware of `rustfmt.toml` and automatically picks it up to use it for configuration, so if the following diff is made to the `rustfmt.toml`:
```patch
@@ -1,7 +1,7 @@
 # Copied from Rust compiler's rustfmt.toml settings

 # Run rustfmt with this config (it should be picked up automatically).
-# unstable_features=true # provided by vargo
-# version=Two # provided by vargo
+unstable_features = true
+version = "Two"
 use_small_heuristics = "Max"
 merge_derives = false
```
then Verusfmt is able to successfully pick that information up and use it.

@utaal: do you know why you are passing it as CLI arguments and keeping it commented-out in the `rustfmt.toml`? From a quick test with the above diff, I see it successfully being picked up both by rustfmt as well as verusfmt, so unless there is a good reason to not use it in the TOML, can we switch it over to only be in the TOML?

In particular the following bits of code would need to be updated:
https://github.com/verus-lang/verus/blob/f2bf563b9f8995b539da4dab2d07ac5f3442c65b/source/rustfmt.toml#L4-L5 and https://github.com/verus-lang/verus/blob/f2bf563b9f8995b539da4dab2d07ac5f3442c65b/tools/vargo/src/main.rs#L764-L765

I'll open a PR shortly with the changes

Reply:
Haha, just ran a full test, and found out what the issue is: it requires nightly only to be able to place it into the `.toml`, but it worked for my separate test due to my local config. Going to have to think about this a little bit.

Reply:
Funnier: support for `version=Two` is gated under `unstable_features=true` which is gated under nightly. For at least 4 years, people have been complaining about "allow us to pass `unstable_features=true`" and the stance seems to be that there should be literally no way for a stable rustfmt to have support for unstable features in their config, at least until rustfmt 2.x comes out; latest post on that ~4 year old thread is from ~2 years ago saying "Sorry to necrobump, but the mythical rustfmt version 2.0 seems not to be released as of March 2022. Is it possible to estimate how long is it going to finally release 2.0 to the public?", so I'm not holding out much hope.

Reply:
Fwiw, the _specific_ example of the `line_count` tool itself I think is not an important one to worry about, since it is isn't actually Verus code meant to be formatted with `verusfmt`, so this _particular_ example should not matter.  Literally, verusfmt looks at it, decides that it has no `verus! { ... }` code, so just invokes rustfmt directly, which picks up the `rustfmt.toml` config, and runs with it.

For most software, I don't actually expect them to be using `version=Two` on a stable Rust (I'm somewhat surprised we're using it; if we were on nightly, we wouldn't have had this issue, and the toml would work just fine).

If we remove `version=Two` from Verus entirely, the diff isn't _massive_; just some parts of the code (mostly if-else blocks) become more spread out across lines, rather than single-line.  Something to maybe consider if we find the disagreement to be an issue.  The other option is that I could expose a `--rustfmt-config` flag within Verusfmt for passing CLI arguments of the kind that rusfmt would take.  Yet another option is for such situations to run verusfmt with `--verus-only` and just use `rustfmt` as it expects to do so on the file, as a separate tool, rather than attempting to have Verusfmt handle everything.